### ISSUE TITLE
Snackbar examples disappear with no ability to adjust timing 

### ISSUE THREAD

**Issue by mfairchild365** [angular/material2#15043]
#### What is the expected behavior?
Users should have enough time to find and interact with the example snackbar content before it disappears.

#### What is the current behavior?
Users might not have enough time to find and interact with the example snackbar content before it disappears.

#### What are the steps to reproduce?
1. Open the [snackbar example](https://material.angular.io/components/snack-bar/examples)
2. Click the "pizza party" button
3. Observe that the snackbar is displayed for about 3 seconds before being removed

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
- Angular Material Example 7.2.2 located at https://material.angular.io/components/
- NVDA 2018.4.1 + FireFox 64
- VoiceOver for IOS (12.1.3) + Safari (12.1.3)

#### Is there anything else we should know?
With the current timeout, users might not be able to find the content when using a screen magnifier before it disappears.

A few possible solutions might be to:
1. add a mechanism to the example that lets users adjust the timeout
2. require that users acknowledge the snackbar before it disappears

This issue should also be discussed in the accessibility section of the documentation for the component.
